datasets/za/pmg_legislators/crawler.py

```diff
@@ -0,0 +1,165 @@
+
+
+def parse_person(context: Context, data: dict):
+    person_id = data.get("id")
+    if not person_id:
+        context.log.error(f"Missing ID in person data: {data}")
+        return
+    person = context.make("Person")
+    person.id = person_entity_id(context, person_id)
+    person.add("name", data.get("name"))
+    person.add("alias", [o.get("name") for o in data.get("other_names", [])])
+    person.add("gender", data.get("gender"))
+    person.add("firstName", data.get("given_name"))
+    person.add("lastName", data.get("family_name"))
+    person.add("birthDate", data.get("birth_date"))
+    person.add("deathDate", data.get("death_date"))
+    person.add("notes", data.pop("summary", None))
+    person.add("title", data.pop("honorific_prefix", None))
+    person.add("topics", "role.pep")
+
+    for link in data.pop("links", []):
+        url = link.get("url")
+        if link.get("note") in ("website", "blog", "twitter", "facebook"):
+            person.add("website", url)
+
+    for contact_detail in data.pop("contact_details", []):
+        value = contact_detail.get("value")
+        if "email" == contact_detail.get("type"):  # might want to remove later
+            person.add("email", clean_emails(value))
+        if "phone" == contact_detail.get("type"):
+            person.add("phone", clean_phones(value))
+        if "cell" == contact_detail.get("type"):
+            person.add("phone", clean_phones(value))
+        if "address" == contact_detail.get("type"):
+            person.add("address", value)
+        if "postal_address" == contact_detail.get("type"):
+            person.add("address", value)
+
+    for ident in data.pop("identifiers", []):
+        identifier = ident.get("identifier")
+        scheme = ident.get("scheme")
+        if scheme == "wikidata" and identifier.startswith("Q"):
+            person.add("wikidataId", identifier)
+
+    context.emit(person, target=True)
+    return person
+
+
+def parse_membership(
+    context: Context, data: dict, organizations: Dict[str, str]
+) -> Optional[str]:
+    person_id = data.get("person_id")
+    if not person_id:
+        context.log.error("Missing person_id in membership data.")
+        return None
+    org_id = data.get("organization_id")
+    org_name = organizations.get(org_id)
+    if not org_name:
+        context.log.error(f"Organization with ID {org_id} not found.")
+        return None
+    org_id = data.get("organization_id")
+    role = data.get("role")
+    role = data.get("role")
+    if role is None:
+        # context.log.error("Role is missing in membership data.")
+        return None
+
+    # Check if the role matches the positions of interest
+    if not any(re.findall(position, role) for position in POSITIONS_OF_INTEREST):
+        context.log.info(f"Skipping role {role} not of interest.")
+        return None
+
+    position_property = post_summary(
+        org_name, role, [data.get("start_date")], [data.get("end_date")], []
+    )
```

Comment:
cool find!

It's probably fine to keep that, but what we really want in opensanctions is a Position made using the make_position helper (with is_pep=True because we already filter for valid positions in the code), and an Occupancy made using the make_occupancy helper. Passing in the birth, death, and position dates, will determine the status and whether this person is still a PEP or not.

https://zavod.opensanctions.org/peps/